### PR TITLE
[DOCFIX] Perf tuning doc

### PR DESCRIPTION
**Pull request by aaudiber**
@apc999 @madanadit @gpang Please list any other perf tuning tips you can think of

### CODE REVIEW

File: docs/en/advanced/Performance-Tuning.md [hunk 1]
```diff
@@ -5,7 +5,328 @@ nickname: Performance Tuning
...
+
+## General Tuning
+
+### JVM Monitoring
+
+To detect long GC pauses, Alluxio administrators can set
+"alluxio.master.jvm.monitor.enabled=true"
+for masters or "alluxio.worker.jvm.monitor.enabled=true"
+for workers. When this is activated, Alluxio will run a monitoring thread which periodically runs
+Thread.sleep and tests the delay between when it was supposed to wake up and when it actually woke
```
Comment (apc999) [inline docs/en/advanced/Performance-Tuning.md]: nit: <pre>`Thread.sleep`</pre>

File: docs/en/advanced/Performance-Tuning.md [hunk 2]
```diff
@@ -5,7 +5,328 @@ nickname: Performance Tuning
...
+input data at about the same time (e.g., to start a SparkSQL query, all Spark executors may read the
+same paquet header); as a result, the same block may be cached on every node. This can be a waste of
+UFS bandwidth as all workers are reading the same data from UFS, as well as the Alluxio storage
+capacity is not effectively used as excessive copies of the same data may evict other valuable
+cached data., and may lead to much worse performance when the connection UFS is slow and Alluxio
+space is almost full.
+
+This performance degradation is due to less-optimized UFS read schedule which is termed as the
+“thundering-herd" problem in Alluxio. One way to resolve this problem is to apply deterministic
+hashing. By specifying “alluxio.user.ufs.block.read.location.policy=DeterministicHashPolicy",
```
Comment (apc999) [inline docs/en/advanced/Performance-Tuning.md]: "  -> `

File: docs/en/advanced/Performance-Tuning.md [hunk 3]
```diff
@@ -5,7 +5,328 @@ nickname: Performance Tuning
...
+Alluxio will select only one “random" worker to read the given block from UFS. As a result, even
+multiple Alluxio clients are requesting the same block from different nodes, only that “selected"
+worker will read from UFS and cache only on that worker. Sometimes we want more parallelism, then
+one can also change the value of
+“alluxio.user.ufs.block.read.location.policy.deterministic.hash.shards" from 1 (default) to N so
```
Comment (apc999) [inline docs/en/advanced/Performance-Tuning.md]: “ -> `

File: docs/en/advanced/Performance-Tuning.md [hunk 4]
```diff
@@ -5,7 +5,328 @@ nickname: Performance Tuning
...
+consider reducing this value so that checkpoints happen more often.
+
+### UFS block locations cache
+
+Alluxio provides block locations, similar to the HDFS client. If a file block is not in Alluxio
+storage yet, the Alluxio will still return the UFS block locations. Therefore, if a file block is
+not in Alluxio storage, the Alluxio master will have to consult the UFS for the UFS block locations,
+which would require an additional RPC to the UFS. This extra overhead may be expensive, so the
+Alluxio master maintains a cache to store these UFS block locations. The parameter
+alluxio.master.ufs.block.location.cache.capacity controls the size of this cache. If it is set to 0,
```
Comment (apc999) [inline docs/en/advanced/Performance-Tuning.md]: quote

File: docs/en/advanced/Performance-Tuning.md [hunk 5]
```diff
@@ -5,7 +5,328 @@ nickname: Performance Tuning
...
+alluxio.master.ufs.path.cache.threads. Increasing the number of threads can decrease the staleness
+of the UFS path cache, but may impact performance by increasing work on the Alluxio master, as well
+as increasing the parallel interactions with the UFS. If this is set to 0, the cache is disabled,
+and the "Once" setting will behave like the "Always" setting.
+
+## Worker Tuning
+
+### Block thread pool size
+
+Alluxio.worker.block.threads.max configures the maximum number of incoming RPC requests to block
```
Comment (apc999) [inline docs/en/advanced/Performance-Tuning.md]: quote `Alluxio.worker.block.threads.max`

File: docs/en/advanced/Performance-Tuning.md [hunk 6]
```diff
@@ -5,7 +5,328 @@ nickname: Performance Tuning
...
+worker that can be handled. This value is used to configure maximum number of threads in Thrift
+thread pool with block worker. This value should be greater than the sum of
+alluxio.user.block.worker.client.threads across concurrent Alluxio clients. Otherwise, the worker
```
Comment (apc999) [inline docs/en/advanced/Performance-Tuning.md]: quote `alluxio.user.block.worker.client.threads`

File: docs/en/advanced/Performance-Tuning.md [hunk 7]
```diff
@@ -5,7 +5,328 @@ nickname: Performance Tuning
...
+
+A tunable configuration for this feature is the
+“alluxio.worker.network.netty.async.cache.manager.threads.max" property. This determines the maximum
+number of concurrent blocks a worker will attempt to cache. By default the value is high (512), but
+in cases where large amounts of data are expected to be asynchronously cached concurrently, lowering
+the value can reduce resource contention (e.g., # of cores)
+
+### Netty Configs
+
+alluxio.worker.network.netty.rpc.threads.max - Default value is 2048, consider increasing this value
```
Comment (apc999) [inline docs/en/advanced/Performance-Tuning.md]: quote alluxio.worker.network.netty.rpc.threads.max

File: docs/en/advanced/Performance-Tuning.md [hunk 8]
```diff
@@ -5,7 +5,328 @@ nickname: Performance Tuning
...
+limit the frequency of UFS syncing. Larger values will perform UFS sync less, but the chances will
+be higher for the Alluxio metadata to be stale. Interacting with the UFS incurs performance
+overheads, so it is recommended to limit frequent UFS syncing.
+
+Since the UFS sync feature includes the loading metadata functionality, if UFS syncing is performed,
+the loading metadata configuration parameter, alluxio.user.file.metadata.load.type, is ignored. In
+comparison to the loading metadata feature, UFS sync takes care of any UFS modifications, like
+deleting files, or updating the timestamp, in addition to discovering new files.
+
+### Passive caching
```
Comment (apc999) [inline docs/en/advanced/Performance-Tuning.md]: I remember that we decided not to talk about this @calvinjia , can you recall?

**Comment by aaudiber** [comment]
@calvinjia can you suggest any other perf tuning tips?

**Comment by apc999** [comment]
Added a step in checking cluster health and a step on checking GC

File: docs/en/advanced/Performance-Tuning.md (hunk 8 above)
Comment (calvinjia) [reply docs/en/advanced/Performance-Tuning.md]: Passive caching is something we hope to remove in the future.

**Pull request merged**